runtime/gc_structs/ClassLoaderClassesIterator.hpp

```diff
@@ -62,6 +62,15 @@ class GC_ClassLoaderClassesIterator
 		ANONYMOUS_CLASSES
 	};
 	ScanModes _mode; /**< indicate type of classes to be iterated */
+	J9Class *_iterateArrayClazz;
+	J9Class *_startingClass;
+	int _arrayState;
+	enum {
+		classLoaderClassIterator_state_valueTypeArray = 0,
+		classLoaderClassIterator_state_valueTypeArrayList,
+		classLoaderClassIterator_state_arrayList,
+		classLoaderClassIterator_state_done
+	};
```

Comment:
You can use named enum and use this name to define variable:

    enum ArrayClassState {
        classLoaderClassIterator_state_valueTypeArray = 0,
        classLoaderClassIterator_state_valueTypeArrayList,
        classLoaderClassIterator_state_arrayList,
        classLoaderClassIterator_state_done
    };
    ArrayClassState _arrayState;

Also all class members should be initialized in Constructor.

Reply:
BTW state names are private, you can skip `classLoaderClassIterator` prefix if you like